File: data_structures_by_python/linked_list/circular_linked_list/insert_node.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None


def insert_node(val):
    global head
    newnode = Node(val)
    if head is None:
        head = newnode
        newnode.next = head
        return
    tmp = head
    while tmp.next != head:
        tmp = tmp.next
    tmp.next = newnode
    newnode.next = head


def traversal():
    if head is None:
        return
    tmp = head
    while True:
        print(tmp.val, end=" ")
        tmp = tmp.next
        if tmp == head:
            break
    print()


head = None

insert_node(10)
insert_node(20)
insert_node(30)
insert_node(40)

traversal()

File: data_structures_by_c++/linked_list/circular_linked_list/insert_node.cpp
#include <bits/stdc++.h>
using namespace std;

class Node
{
public:
    int val;
    Node *next;

    Node(int val)
    {
        this->val = val;
        this->next = NULL;
    }
};

void insert_node(Node *&head, int val)
{
    Node *newnode = new Node(val);
    if (head == NULL)
    {
        head = newnode;
        newnode->next = head;
        return;
    }
    Node *tmp = head;
    while (tmp->next != head)
    {
        tmp = tmp->next;
    }
    tmp->next = newnode;
    newnode->next = head;
}

void traversal(Node *head)
{
    if (head == NULL)
        return;
    Node *tmp = head;
    while(true)
    {
        cout << tmp->val << " ";
        tmp = tmp->next;
        if (tmp == head)
            break;
    }
    cout << endl;
}

int main()
{
    Node *head = NULL;

    insert_node(head, 10);
    insert_node(head, 20);
    insert_node(head, 30);
    insert_node(head, 40);

    traversal(head);
}
